src/main/java/org/sonarlint/intellij/core/SonarQubeEventNotifications.java

```diff
@@ -0,0 +1,140 @@
+    this.busConnection = project.getMessageBus().connect(myProject);
+  }
+
+  @Override
+  public void projectOpened() {
+    register(projectSettings);
+    busConnection.subscribe(ProjectConfigurationListener.TOPIC, settings -> {
+      // always reset notification date, whether bound or not
+      projectState.setLastEventPolling(ZonedDateTime.now());
+      register(settings);
+    });
+  }
+
+  private void register(SonarLintProjectSettings settings) {
+    SonarQubeNotifications.get().remove(eventListener);
+    SonarQubeServer server;
+    try {
+      server = bindingManager.getSonarQubeServer();
+    } catch (InvalidBindingException e) {
+      // do nothing
+      return;
+    }
+    if (settings.isBindingEnabled() && server.enableNotifications()) {
+      NotificationConfiguration config = createConfiguration(settings, server);
+      SonarQubeNotifications.get().register(config);
+    }
+  }
+
+  private NotificationConfiguration createConfiguration(SonarLintProjectSettings settings, SonarQubeServer server) {
+    String projectKey = settings.getProjectKey();
+    ServerConfiguration serverConfiguration = SonarLintUtils.getServerConfiguration(server);
+    return new NotificationConfiguration(eventListener, notificationTime, projectKey, serverConfiguration);
+  }
+
+  /**
+   * Read and save directly from the mutable object.
+   * Any changes in the project settings will affect the next request.
+   */
+  private class ProjectNotificationTime implements SonarQubeNotificationTime {
+
+    @Override public ZonedDateTime getLastNotificationTime() {
+      ZonedDateTime lastEventPolling = projectState.getLastEventPolling();
+      if (lastEventPolling == null) {
+        lastEventPolling = ZonedDateTime.now();
+        projectState.setLastEventPolling(lastEventPolling);
+      }
+      return lastEventPolling;
+    }
+
+    @Override public void setLastNotificationTime(ZonedDateTime dateTime) {
+      // this could be false if the settings changed between the read and write
+      if (dateTime.isAfter(projectState.getLastEventPolling())) {
+        projectState.setLastEventPolling(dateTime);
+      }
+    }
+  }
+
+  /**
+   * Simply displays the events and discards it
+   */
+  private class EventListener implements SonarQubeNotificationListener {
+    @Override public void handle(SonarQubeNotification notification) {
+      Notification notif = new Notification(GROUP_SONARQUBE_EVENT,
+        "<b>SonarLint - SonarQube event</b>",
+        createMessage(notification),
+        NotificationType.INFORMATION,
+        new NotificationListener.UrlOpeningListener(true));
+      notif.setImportant(true);
+      notif.notify(myProject);
+    }
+
+    private String createMessage(SonarQubeNotification notification) {
+      return notification.message() + ". Click <a href=\"" + notification.link() + "\">here</a> to open in browser.";
+    }
+  }
+
+  @Override
+  public void projectClosed() {
```

Comment:
I would move this next to `projectOpened`, because I think it's easier to read when symmetric operations are close to each other